Frameworks/CoreGraphics/D2DWrapper.mm

```diff
@@ -0,0 +1,33 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <CoreGraphics/D2DWrapper.h>
+#import <Starboard.h>
+
+using namespace Microsoft::WRL;
+
+// Private helper for creating a D2DFactory
+static ComPtr<ID2D1Factory> __CreateD2DFactoryInstance() {
+    ComPtr<ID2D1Factory> d2dFactory;
+    THROW_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_MULTI_THREADED, __uuidof(ID2D1Factory), &d2dFactory));
```

Comment:
This should be D2D1_FACTORY_TYPE_SINGLE_THREADED.  We dont need any locking from D2D as all our drawing is on UI thread.  Specifying multi threaded here is counter intuitive and will result in worse performance.  Note: even if the factory is single threaded, when we have hardware offloading, D2D will parallelize the tasks on the GPU.  https://msdn.microsoft.com/en-us/library/windows/desktop/dd368104(v=vs.85).aspx has all the details.